Assets/Shopify/UIToolkit/ShopControllers/Classes/CartController.cs

```diff
@@ -38,15 +41,20 @@ public class CartItemsChangeEvent : UnityEvent<List<CartItem>> {}
         public UnityEvent OnPurhcaseCancelled = new UnityEvent();
         public PurchaseFailedEvent OnPurhchaseFailed = new PurchaseFailedEvent();
 
-        private Dictionary<string, ProductVariant> _idsToVariants = new Dictionary<string, ProductVariant>();
+        private struct ProductAndVariantPair {
+            public Product product;
+            public ProductVariant variant;
+        }
+
+        private Dictionary<string, ProductAndVariantPair> _idsToProductPairs = new Dictionary<string, ProductAndVariantPair>();
 
         public CartController(Cart cart) {
             SetCart(cart);
         }
 
         public void SetCart(Cart cart) {
             Cart = cart;
-            _idsToVariants = new Dictionary<string, ProductVariant>();
+            _idsToProductPairs = new Dictionary<string, ProductAndVariantPair>();
```

Comment:
Would `Clear` https://msdn.microsoft.com/en-us/library/b5txwy7s(v=vs.110).aspx better communicate the intent here?